storage/merkle_path.go

```diff
@@ -0,0 +1,73 @@
+package storage
```

Comment:
Yeah, but this is mostly doing merkle related calculations and then representing the output of that as a bunch of `storage.NodeID`s.  The SMT writer code (in also in `merkle/`) has a dep on `storage.NodeID` too, since it needs to calculate NodeIDs to look them up.